chainstate/src/detail/tx_verification_strategy/default_strategy.rs

```diff
@@ -115,6 +106,15 @@ impl TransactionVerificationStrategy for DefaultTransactionVerificationStrategy
             .check_block_reward(block, Fee(total_fees), Subsidy(block_subsidy))
             .log_err()?;
 
+        let reward_fees = tx_verifier
```

Comment:
Why move this here?

Reply:
`connect_transactable` spends the kernel, as a result, `check_block_reward` won't pass. So I switched the order.  It was not a problem before because we didn't have inputs in block reward

Reply:
I see. The issue here is that we should verify the reward before getting into the block because the reward is the key part of a block's validity. Maybe throw in a TODO here and explain this problem.

Reply:
I think we can separate txs checks and fee collection. The only downside is that we will have to enumerate and perform some calculations twice. I'll place a TODO to look into this later